src/Minsk/CodeAnalysis/Lowering/Lowerer.cs

```diff
@@ -263,12 +222,145 @@ protected override BoundStatement RewriteConditionalGotoStatement(BoundCondition
                 var condition = (bool)node.Condition.ConstantValue.Value;
                 condition = node.JumpIfTrue ? condition : !condition;
                 if (condition)
-                    return RewriteStatement(new BoundGotoStatement(node.Label));
+                    return RewriteStatement(Goto(node.Label));
                 else
-                    return RewriteStatement(new BoundNopStatement());
+                    return RewriteStatement(Nop());
             }
 
             return base.RewriteConditionalGotoStatement(node);
         }
+
+        #region BoundNodeFactory
+
+        private BoundNopStatement Nop()
+        {
+            return new BoundNopStatement();
+        }
+
+        private BoundLabelStatement Label()
+        {
+            var label = GenerateLabel();
+            return new BoundLabelStatement(label);
+        }
+
+        private BoundLabelStatement Label(BoundLabel label)
+        {
+            return new BoundLabelStatement(label);
+        }
+
+        private BoundLiteralExpression Literal(object literal)
+        {
+            Debug.Assert(literal is string || literal is bool || literal is int);
+
+            return new BoundLiteralExpression(literal);
+        }
+
+        private BoundBlockStatement Block(params BoundStatement[] stmts)
+        {
+            return new BoundBlockStatement(ImmutableArray.Create(stmts));
+        }
+
+        private BoundGotoStatement Goto(BoundLabelStatement label)
+        {
+            return new BoundGotoStatement(label.Label);
+        }
+
+        private BoundGotoStatement Goto(BoundLabel label)
+        {
+            return new BoundGotoStatement(label);
+        }
+
+        private BoundConditionalGotoStatement Goto(BoundLabelStatement label, BoundExpression condition, bool jumpIfTrue)
+        {
+            return new BoundConditionalGotoStatement(label.Label, condition, jumpIfTrue);
+        }
+
+        private BoundConditionalGotoStatement GotoTrue(BoundLabelStatement label, BoundExpression condition)
+            => Goto(label, condition, jumpIfTrue: true);
+
+        private BoundConditionalGotoStatement GotoFalse(BoundLabelStatement label, BoundExpression condition)
+            => Goto(label, condition, jumpIfTrue: false);
+
+        private BoundVariableDeclaration VariableDeclaration(VariableSymbol symbol, BoundExpression initExpr)
+        {
+            return new BoundVariableDeclaration(symbol, initExpr);
+        }
+
+        private BoundVariableExpression Variable(VariableSymbol symbol)
+        {
+            return new BoundVariableExpression(symbol);
+        }
+
+        private BoundVariableExpression Variable(BoundVariableDeclaration varDecl)
+        {
+            return new BoundVariableExpression(varDecl.Variable);
+        }
+
+        private BoundVariableDeclaration ConstantDeclaration(string name, BoundExpression initExpr, TypeSymbol? type = null)
```

Comment:
Should match the names of `BoundVariableDeclaration`, so `initExpr` should be `initializer`